### ISSUE TITLE
zoomed_inset_axes places the inset box outside the figure when the axes are inverted

### ISSUE THREAD

**Issue by jieunchoi** [matplotlib/matplotlib#8589]
### Bug report

**Bug summary**

Now when using mpl_toolkits.axes_grid1.inset_locator.zoomed_inset_axis with an inverted axis, the inset box gets placed outside the figure. This was not an issue in the past.

**Code for reproduction**

This is okay (image B):
```python
from mpl_toolkits.axes_grid1.inset_locator import zoomed_inset_axes, mark_inset
fig = plt.figure()
ax = plt.subplot(111)
ax.plot([1,2,3], [1,2,3])
ax.axis([1,3,1,3])
inset_ax = zoomed_inset_axes(ax, zoom=2.5, loc=4)
inset_ax.axis([1.1, 1.4, 1.1, 1.4])
```

but this is not (image A):
```python
from mpl_toolkits.axes_grid1.inset_locator import zoomed_inset_axes, mark_inset
fig = plt.figure()
ax = plt.subplot(111)
ax.plot([1,2,3], [1,2,3])
ax.axis([3,1,3,1])
inset_ax = zoomed_inset_axes(ax, zoom=2.5, loc=4)
inset_ax.axis([1.4, 1.1, 1.4, 1.1])
```

**Actual outcome**
image A
![image](https://cloud.githubusercontent.com/assets/4583333/25784717/e0f7a9ae-333f-11e7-9db8-23632d14629b.png)

**Expected outcome**
image B
![image](https://cloud.githubusercontent.com/assets/4583333/25784714/de5a6df8-333f-11e7-82aa-be6906a10749.png)

**Matplotlib version**

  * Operating System: Mac OS 10.12.4
  * Matplotlib Version: 2.0.0
  * Python Version: 2.7.13
  * Jupyter Version (if applicable):
  * Other Libraries: 

python and matplotlib installed with conda: Anaconda 4.3.1

**Comment by dstansby** [comment]
Fixed by #8601

**Comment by jieunchoi** [comment]
Many thanks to all!
